compiler/qsc_frontend/src/typeck/convert.rs

```diff
@@ -46,11 +43,28 @@ pub(crate) fn ty_from_ast(names: &Names, ty: &ast::Ty) -> (Ty, Vec<MissingTyErro
                 Some(&resolve::Res::Item(item)) => Ty::Udt(hir::Res::Item(item)),
                 Some(&resolve::Res::PrimTy(prim)) => Ty::Prim(prim),
                 Some(resolve::Res::UnitTy) => Ty::Tuple(Vec::new()),
-                Some(resolve::Res::Local(_)) | None => Ty::Err,
+                // a path should never resolve to a parameter,
+                // as there is a syntactic difference between
+                // paths and parameters.
+                // So realistically, by construction, `Param` here is unreachable.
+                Some(resolve::Res::Local(_) | resolve::Res::Param(_)) => unreachable!(
+                    "A path should never resolve \
+                    to a local or a parameter, as there is syntactic differentiation."
```

Comment:
FYI:
* I have an impression that some noun is missing between `local` and `or` in green line 52 (L52).
* L52 ends with dot `.` (in front of `"`), but L64 does not.
I personally prefer to not end messages with `.` because some later stage may want to take that message into the parentheses or quotes or angle brackets, etc. Dot in such contexts looks redundant.

Reply:
It's common to use "local" and "global" as shorthand for "local variable" and "global variable." Saying "a local" or "the locals" sounds fine to me.